### DISCUSSION TITLE
Zappi integration

### DISCUSSION THREAD

**Discussion by yebaws** [CJNE/ha-myenergi#457] [General]
Interested to know peoples experience of the myenergi Zappi integartion. 

We're buying an EV and currently looking at chargers. Having a useful HA integration is essential. 

For example, would I be able to make an automation start charging when my home storage batteries are full, and excess solar PV is starting to be exoprted....and just charge at the rate that uses up the excess solar? 

I read that Zappi can integrate with solar PV with it's own CT clamp. Running a cable from the charger to the PV cable will be very difficult, but I do have all the data from solar generation and whole house import / export in sensors from other CT clamps, which I could use to control the Zappi if the API will let me?

**Comment by G6EJD** [comment]
All that you have described is done by the Zappi without any programming or need for an API, it uses a CT on the Grid supply to detect solar PV export and has controls to prevent battery discharge, including timers to alter priorities.

**Reply by yebaws** [reply]
Sure, but because of the charger's position, I can't get the hard-wired CT clamp from the Zappi to the grid supply, although there is a Shelly CT clamp on the incoming main, so I do have a sensor with that import / export data. Which is why I would like to be able to control the Zappi''s behaviour with automations...

**Reply by LeiChat** [reply]
You could add myenergi's harvi.
https://www.myenergi.com/news/the-harvi/

**Reply by yebaws** [reply]
Thanks for that suggestion. Do you know what wireless protocol the harvi uses? Google says up to 25m, which would be within the distance between the CU and the charger, but one external (SIP panel) wall to go through as well. Could maybe test it with something else if I knew what protocol it used.

**Comment by G6EJD** [comment]
‘Running a cable from the charger to the PV cable will be very difficult‘
The confusion is you said solar pv, it needs to be the grid. You can run cable lengths up to 100M

**Reply by yebaws** [reply]
sorry, yes, no gain from knowing how much PV is being produced unless you know when export starts. Distance not as much the issue as terrain / walls etc. I'm guessing that as nobody has said "yes you can do it" that HA control over the Zappi is somewhat limited. I really need a "dumb" charge that can be fully integrated and will do whatever it is told to do. So far no luck finding this.

**Comment by G6EJD** [comment]
The protocol is proprietary and unpublished but uses chirp modulation derived from an RFM69 chip, it will be based on LoRa and uses the 868MHz band, but the licence requirements are for very low power and therefore range is limited, maybe 20M maximum. 
the MyEnergi integration does allow starting and stoping charging (well switching the output on/off as the EV does the charging) but to meet safety requirements you really do need the grid CT. Using a HA automation you could use another entity like a Shelly Grid CT with an automation, but then the Zappi would be a simple EVSE and not worth the investment. But few EVSE’s offer API access you you may find it difficult to find another solution.

**Reply by yebaws** [reply]
A bit "sneaker network", but my best option might be to plug in a granny charger during the day and control that with HA and a smart plug

**Comment by plord12** [comment]
If you are in the UK, zappi is now supported by Octopus Intelligent Go.  In this case, Octopus controls the zappi and they choose the specific charging "slots" when you plug the car in (based on your requirements).  These slots are at 7.5p/kWh which can be used for other purposes.

Also, the Octopus export prices are now pretty good ... ie you're better off exporting excess solar (15p/kWh) and charging your car overnight (7.5p/kWh).  

In otherwords, not much point in using HA to optimize charging your car with Octopus intelligent.

But, linking the Octopus Intelligent slots to your battery is still worth while (IMHO) ... eg charge your house battery at these lower rates.  Its also worth playing the game with "saving sessions" ... ie charge your battery before the saving session starts (at full grid price) and export it to the grid during the session ... this pays you £2.25/kWh exported (!).

So for me, using HA to only monitor car charging makes most sense, but also trying to sync my libbi to make best use of the dynamic Octopus rates.

**Reply by yebaws** [reply]
Thanks for that thought.

Unfortunately, I self installed my solar which, for reasons best known to our government, prevents me from receiving any export payments from my PV at the moment. And my home charging is self installed too, so the same restriction exists. Slightly off-topic, but from what I read on the Octopus website, I will actually be able to register for an export tarriff once I have an EV, which would give me a roundabout way of getting payments for PV exports. Personally, I think that everyone should be paid for what they export - by law.

**Reply by gilesD1** [reply]
Thanks for the information.  I signed up for Octopus Intelligent Go after a Zappi install and saw my costs double.  When I enquired with Octopus they informed me that **only** the power supplied via the Zappi to the car was metered at 7.5p / kWh.  The house including batteries is metered at the rate of 30.75p / kWh at all times.  Can you give any insight into how you have configured (or wired) your house batteries to benefit from the 7.5p rate please?

**Reply by G6EJD** [reply]
It's impossible for Octopus to know whether your charging an EV or have the Kettle on, all they can do is estimate what energy was put into your EV and charge you for that. There is no wiring method that can be employed to do what you need.

**Comment by AJediIAm** [comment]
Zappi does not require CT clamps at the solar installation. It can work fine with only CT clamps on the main power line.

Alternatives are controlling the charge rate through your car or switch the charger on or off through a 3-phase relay like the Shelly pro 3.

**Comment by plord12** [comment]
> Thanks for the information. I signed up for Octopus Intelligent Go after a Zappi install and saw my costs double. When I enquired with Octopus they informed me that **only** the power supplied via the Zappi to the car was metered at 7.5p / kWh. The house including batteries is metered at the rate of 30.75p / kWh at all times. Can you give any insight into how you have configured (or wired) your house batteries to benefit from the 7.5p rate please?

Huh.  I don't see that.  See https://octopus.energy/smart/intelligent-octopus-go/ 

> Automatically charge your car when it's cheapest and get super low smart charging rates plus 6h of cheap energy for your whole home every night.

Eg my house battery is charged during the 6h (and any additional slots )

This is an example from my bill on a charging day ... 

<img width="530" alt="Screenshot 2024-01-25 at 17 12 11" src="https://github.com/CJNE/ha-myenergi/assets/63543143/a24f8182-6a76-4056-ae6a-facb4c977259">

My battery is a myenergi libbi, BTW, which is configured with a changing schedule -

<img width="1157" alt="Screenshot 2024-01-25 at 17 15 16" src="https://github.com/CJNE/ha-myenergi/assets/63543143/e1c9bbb7-6852-4a50-bbd7-becc3e7a9846">

I have a hack that mirrors the octopus schedule (inc slots) to the myenergi libbi schedule when it gets updated.

**Reply by gilesD1** [reply]
Many thanks for the information which supports my original understanding and motivation for installing the Zappi.  The response I received from Octopus when I queried the increase in cost, which equated to their day rate for a night time charge is

_On Wed, 24 Jan 2024 at 08:45, Octopus Energy hello@octopus.energy wrote:

Hi Giles,

Thank you for reaching out to us. I apologise for any confusion caused.

The 23:30 - 05:30 rate of 7.5p / kWh applies specifically to the charging
of your Zappi charging point. The rest of your property will be supplied at
the standard rate of 30.25p / kWh.

If you have any further questions or need assistance, please let me know.

Kind regards,_

I have written again for clarification as either their description is inaccurate or they didn't get me onto their tariff correctly

**Reply by LeiChat** [reply]
Apologies if you are already aware. The idea of IO Go is that you give Octopus control of the zappi boosting in exchange for them billing your consumption at the off-peak rate during any scheduled boost they setup. It does not work the other way, you cannot just boost the zappi manually at any time and expect that consumption to be billed at the off-peak rate.

In theory, Octopus could bill you for the zappi consumption based on data from myenergi's API but, for now, asG6EJD said, they just bill based on their scheduled boost slots, regardless of what is consuming electricity in your home. They schedule boosts based on the % and finish time you specify.

The advice in the email you've shared from Octopus is incorrect. (Unless they've changed things and not informed any of us.)

There's a default off-peak period (for your home) and you can have additional off-peak rate boost slots allocated by Octopus based on what you configure in their app.

**Reply by yebaws** [reply]
@gilesD1 
What Octopus have told you is incorrect - not unusual in my experience.

see here: https://octopus.energy/intelligent-go-faqs/

"Intelligent Octopus Go is a smart tariff with peak and off-peak pricing. Customers benefit from a low price night rate 
 for their EV and home energy use of 7.5p/kWh between 11:30pm and 5:30am. Outside these hours your unit rate will vary by region."

You just need to make sure your home batteries charge between 23:30 and 05:30. And preferably your car too. Charge your car or use any other electricity outside those hours and it will be at peak rate.

The one exception that Octopus specify, is that when using intelligent charging, they will occasionally charge using a slot outside the 23:30 - 05:30 slot., but they say they will only charge you 7.5p per Kwh if they do this

**Reply by plord12** [reply]
> Many thanks for the information which supports my original understanding and motivation for installing the Zappi. The response I received from Octopus when I queried the increase in cost, which equated to their day rate for a night time charge is

What does your bill say ?  You should see the tariff and the per 1/2 hour rates in the graphs they supply.

**Reply by gilesD1** [reply]
Thank you all for the input and apologies if this is off topic, I'm sure you'll advise.  Here's the background to how I got here.

I've been an Octopus customer for a while now.  I moved my Economy 7 tariff over to Flexible Octopus when my last supplier went bust.  So from 00:30GMT to 07:30GMT I was receiving electricity at circa 15p / kWh and at other times circa 35p / kWh.  At the end of 2022 I had solar PV and batteries installed.  Batteries charge during the off peak period and discharge during the day.  I also have a hybrid car charging via a PodPoint.  All worked well and I was largely able to run the house at the "off peak" price.

Having done some research it appeared that if I replaced the PodPoint with a Zappi I would be able to benefit from an even lower "off peak" rate of 7.5p / hWh and that at other times Octopus could enable charging at the off peak rate.  So, similar to Flexible Octopus but a lower whole house rate for 6 hours (23:30GMT to 05:30GMT) and the potential for intelligent car charging at other times, controlled by Octopus.  This does seem to be the experience of those who have commented on this forum.

On Tuesday 23rd January 2024 the PodPoint was changed to a Zappi and at about 22:00 I changed tariffs to IOG via the Octopus app.  I also linked my Zappi acct to Octopus and received confirmation that intelligent charging was enabled.  I also reconfigured the house battery charging schedule to the IOG "off peak" times.  I went to bed looking forward to seeing a lower cost on the IHD in the morning.

Wednesday morning, 05:40 and my cost has nearly doubled.  Simple arithmetic tells me I've been charged at the "day" rate of circa 30p / kWh.  I provided Octopus with screenshots of the IHD and Smartmeter display (showing the price per kWh).  I asked them to swap me back to Flexible Octopus on the Wednesday afternoon but no change since then.  The same cost and unit rate on Thursday morning and this morning Friday 26th January.  It seems that the change of tariff has applied the day rate across the 24 hours and that nothing is being supplied at the "night" rate, regardless of tariff.

I will now complain.

So apologies for the long and possibly off topic description, but I was originally looking for HA / Zappi integration experience.

@plord12 thank you for the suggestion to look at the graphs (and the other helpful information), unfortunately I cannot locate that facility, can you point me in the right direction please?

**Reply by plord12** [reply]
I wouldn't trust the in-home display.  Get vour actual bill from https://octopus.energy/ and take a look.  Also check the phone app.  At least verify the tariff you are on.

**Reply by gilesD1** [reply]
There is indeed a significant discrepancy between the Octopus numbers and those shown on the IHD and the Smart meter itself.  It appears that for whatever reason the IOG day rate is being used (and displayed) 24 hours a day.  No further response from Octopus so far.

**Reply by gilesD1** [reply]
Thank you, see below, I shall trust the app and not the meter or IHD.
